### DISCUSSION TITLE
CloudWatch Agent on Bottlerocket?

### DISCUSSION THREAD

**Discussion by manipulator01** [bottlerocket-os/bottlerocket#2050] [Q&A]
Hello,

We recently switched our EKS nodes from AmazonLinux to Bottlerocket.
I would like to know how can I install **cloudwatch agent on my bottlerocket instance to collect node level metrics such as RAM.**

Please note that we already run cloudwatch as a daemon to collect logs and metrics from containers. I am asking about collecting host level metrics from bottlerocket

Should I be baking in cloudwatch agent onto my bottlerocket AMI? OR is there a better way?

Could you please point me to right resource?

**Comment by jpculp** [comment]
Hi @manipulator01, thanks for reaching out! In general, we try to keep things isolated from the Bottlerocket host itself. Instead we'd recommend getting that information from Kubernetes, which you can [read more about here](https://kubernetes.io/docs/concepts/cluster-administration/logging/). Another method is to use Fluent Bit as [covered in this blog post](https://aws.amazon.com/blogs/opensource/centralized-container-logging-fluent-bit/).

**Reply by Anorlondo448** [reply]
@jpculp 
Logging can be done with the fluent bit.
How do I get host memory information?
I want to get custom metrics (memory) when using bottlerocket in ECS.

**Reply by jpculp** [reply]
@Anorlondo448 I believe when it comes to ECS the metrics are already logged to CloudWatch. Resource information is collected by the ECS container agent and from there I would think you could monitor it like any other container instance.

**Reply by Anorlondo448** [reply]
@jpculp 
Thank you! 

The default metrics have been obtained.
But can I collect custom metrics? For example, the memory usage of a container instance.

**Reply by ichwill100** [reply]
I agree with @Anorlondo448 .
How can we get custom metrics like memory usage?
Thanks
